test/dataframe.jl

```diff
@@ -325,25 +325,44 @@ module TestDataFrame
     df = DataFrame(a=Union{Int, Missing}[1, 2, 3], b=Union{Float64, Missing}[3.0, 4.0, 5.0])
     @test deleterows!(df, [2, 3]) === df
     @test df == DataFrame(a=[1], b=[3.0])
-
-    # describe
-    #suppress output and test that describe() does not throw
-    devmissing = Compat.Sys.isunix() ? "/dev/null" : "nul"
-    open(devmissing, "w") do f
-        @test nothing == describe(f, DataFrame(a=[1, 2], b=Any["3", missing]))
-        @test nothing ==
-              describe(f, DataFrame(a=Union{Int, Missing}[1, 2],
-                                    b=["3", missing]))
-        @test nothing ==
-              describe(f, DataFrame(a=CategoricalArray([1, 2]),
-                                    b=CategoricalArray(["3", missing])))
-        @test nothing == describe(f, [1, 2, 3])
-        @test nothing == describe(f, [1, 2, 3])
-        @test nothing == describe(f, CategoricalArray([1, 2, 3]))
-        @test nothing == describe(f, Any["1", "2", missing])
-        @test nothing == describe(f, ["1", "2", missing])
-        @test nothing == describe(f, CategoricalArray(["1", "2", missing]))
-    end
+    
+    # Test describe
+    struct CustomStruct end
+    @testset "describe(df)" begin
+        # Construct the test dataframe
+        df = DataFrame(number = [1, 2, 3, 4],
+                       number_missing = [1,2, 3, missing],
+                       non_number = ["a", "b", "c", "d"],
+                       non_number_missing = ["a", "b", "c", missing],
+                       custom_struct = [CustomStruct() for i in 1:4],
```

Comment:
As I said I don't think it's useful to use `CustomStruct` here. Better test `CategoricalArray`.

Reply:
Changed to categorical Array